proxy/context.go

```diff
@@ -218,11 +218,15 @@ func (c *context) ParentSpan() opentracing.Span        { return c.parentSpan }
 func (c *context) Logger() filters.FilterContextLogger {
 	if c.logger == nil {
 		traceId := tracing.GetTraceID(c.initialSpan)
-		if traceId != "" {
-			c.logger = log.WithFields(log.Fields{"trace_id": traceId})
+		if c.proxy != nil && c.proxy.log != nil {
+			c.logger = c.proxy.log
 		} else {
 			c.logger = log.StandardLogger()
 		}
+
+		if traceId != "" {
+			c.logger = c.logger.(*log.Logger).WithFields(log.Fields{"trace_id": traceId})
```

Comment:
@AlexanderYastrebov , what do you think, can I just do type assertion here and assume it will never fail? Or should I better do type switch here instead?

Reply:
`*log.Logger` is `*logrus.Logger`, I would say no

Reply:
Why not add this to filters.FilterContextLogger and then no cast?

Reply:
If you mean adding `WithFields(logrus.Fields)` method to `filters.FilterContextLogger` context, it makes no sense to do this, because this method has argument type specific from logrus. Given that, `filters.FilterContextLogger` interface could not abstract logging libraries other than logrus.

Reply:
Discussed after standup, let's check two options:
1) Hijack test to read logrus output OR
2) Implement `withFields` in `logging.Logger`

Let's try what works in more beautiful way.